src/scrobbler/scrobblemetadata.cpp

```diff
@@ -17,10 +18,56 @@
 *
 */
 
+#include <QString>
+#include <QStringLiteral>
+
 #include "core/song.h"
 
 #include "scrobblemetadata.h"
 
+using namespace Qt::Literals::StringLiterals;
+
+/**
+ * Convert a source to a music service domain name, for ListenBrainz.
+ *
+ * See the "Music service names" note on https://listenbrainz.readthedocs.io/en/latest/users/json.html.
+ */
+static QString serviceFromSource(Song::Source source) {
```

Comment:
Move these functions to `Song` after `IconForFiletype`, since that's where all such functions are.

Reply:
Moved them, would appreciate feedback on whether they're in the right place in the file though (and their naming)